### ISSUE TITLE
Document asymptotic runtime for operations.

### ISSUE THREAD

**Issue by bolinfest** [augustohp/Priority-Queue-NodeJS#5]
Looking at the code, I believe you are using a sorted array as the storage for your queue. This means that `add()` is O(N) in the size of your queue because you might have to shift all of the elements in your array when you do an insertion. From http://pages.cs.wisc.edu/~vernon/cs367/notes/PRIORITY-QAnswers.html:

"If we use an array to store the values in the priority queue, we can either store the values in sorted order (which will make the insert operation slow, and the removeMax operation fast), or in arbitrary order (which will make the insert operation fast, and the removeMax operation slow).

"As mentioned above, if we keep the array sorted, then the insert operation is worst-case O(N) when there are N items in the queue: we can find the place for the new value efficiently using binary search, but then we'll need to move all the larger values over to the right to make room for the new value."

I was trying to find an implementation of a priority queue where insertion is O(lg N), which I believe would be the case if the underlying implementation were a balanced binary search tree. Your implementation has the advantage that finding the min or max value is always O(1), whereas one of those operations would be O(lg N) if you used a BST.

Basically, there are a lot of priority queue implementations for Node out there, and it would be easier to compare if the tradeoffs that were chosen were documented.

**Comment by josephg** [comment]
:+1: 

I think a heap based priority queue should do what you want. I want that too!

**Comment by bolinfest** [comment]
@josephg Agreed: I ended up going with https://www.npmjs.com/package/heap. The API and asymptotic runtime are what I'm looking for. It doesn't have the `capacity` option that this package has, but the work required to maintain the capacity is pretty trivial by encapsulating the `Heap` with another class (e.g., if the `Heap` is already at `capacity` when doing an `insert()`, do a `replace()` instead).

Also, the `heap` package doesn't have any `"dependencies"` listed in its `package.json`, which is also nice.
